tests/foreman/cli/test_realm.py

```diff
@@ -31,6 +31,11 @@
 pytestmark = [pytest.mark.run_in_one_thread]
 
 
+def realm_cleanup(realm_id=None):
```

Comment:
Here its used via `request.addfinalizer`, and you'll want the satellite instance so you can use `target_sat.api.Realm(id=realm_id).delete()`.  Since its getting created in the test and you don't know the ID, I'd just pass the satellite instance in too.

You could use pytest classes here to store `self.realm_id` within the test CALL phase and then write this as a yielding fixture, being able to reference `self.realm_id` instead of depending on the test passing it. Don't think its worth the shuffle though.